backend/typescript/rest/givecloud.ts

```diff
@@ -33,6 +33,16 @@ givecloudRouter.post(
   body("supporter.last_name", "supporter last_name is required").exists(),
   async (req: Request, res: Response) => {
     try {
+      if (process.env.HMAC_SECRET_KEY) {
+        const hash = crypto
+          .createHmac("sha1", process.env.HMAC_SECRET_KEY)
+          .update(JSON.stringify(req.body))
+          .digest("hex");
+        if (hash !== req.get("HTTP_X_GIVECLOUD_SIGNATURE")) {
```

Comment:
Add some spacing here to make this code more readable, add a newline after line 40